cfme/cloud/tenant.py

```diff
@@ -335,7 +365,17 @@ def resetter(self):
 @navigator.register(Tenant, 'Details')
 class TenantDetails(CFMENavigateStep):
     VIEW = TenantDetailsView
-    prerequisite = NavigateToAttribute('parent', 'All')
+
+    def prerequisite(self, *args, **kwargs):
+        """
+        Navigate through provider or collection filter if it exists else navigate through collection
+        object """
+        # Here we assume that every tenant has a parent collection
+        filter = self.obj.provider or self.obj.parent.filters.get('parent')
+        if filter:
+            return navigate_to(filter, 'CloudTenants')
```

Comment:
This navigation destination has only been registered for a `NetworkProvider` parent, but is not registered for a `CloudProvider` parent.  If I have a tenant object with a parent `TenantCollection` that is filtered, or that has a `provider` attribute set, and that provider is a `CloudProvider`, this call with raise `NavigationDestinationNotFound`.

A navigation class must be registered against `CloudProvider`, using the `ProviderTenantAllView`.

Reply:
If Cloud provider would pay so much attention to our Networks provider as we do it for it, then the view would have worked for us out of the box, haha :) I don't mind supporting it for Cloud provider as well, but perhaps we could do it in a followup PR?

Reply:
I can dig that - hooking into existing framework comes with more responsibilities than building the foundations ;)

Ultimately my goal here is to make sure that all the providers live in harmony together, utilizing the same base model/view/navigation code wherever possible.